ocaml/xapi/repository.ml

```diff
@@ -512,50 +512,53 @@ let get_host_updates_in_json ~__context ~installed =
         (ExnHelper.string_of_exn e) ;
       raise Api_errors.(Server_error (get_host_updates_failed, [ref]))
 
+(* Map the URI in the HTTP reuqest to local repo path *)
+let map_uri_to_path ~prefix ~by uri =
```

Comment:
The `by` looks odd here. It could be part of the`root` which is passed to `Helpers.resolve_uri_path`.
So for a file service endpoint, the relative path is generally the string which is trimmed with the endpoint string. The file path would be `root` + `relative path`.